### PR TITLE
Update doc for allowedHeaders

### PR DESCRIPTION
**Pull request by pranavbhole**
Update doc for: https://github.com/apache/druid/pull/16974

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: docs/configuration/index.md [hunk 1]
```diff
@@ -616,9 +616,10 @@ the [HDFS input source](../ingestion/input-sources.md#hdfs-input-source).
 You can set the following property to specify permissible protocols for
 the [HTTP input source](../ingestion/input-sources.md#http-input-source).
 
-|Property|Possible values|Description|Default|
-|--------|---------------|-----------|-------|
-|`druid.ingestion.http.allowedProtocols`|List of protocols|Allowed protocols for the HTTP input source.|`["http", "https"]`|
+|Property| Possible values | Description                                                                                                                               |Default|
+|--------|-----------------|-------------------------------------------------------------------------------------------------------------------------------------------|-------|
+|`druid.ingestion.http.allowedProtocols`| List of protocols | Allowed protocols for the HTTP input source.                                                                                              |`["http", "https"]`|
+|`druid.ingestion.http.allowedHeaders`| A list of permitted request headers for the HTTP input source. By default, the list is empty, which means all headers are allowed in the ingestion specification. |`[]`|
```
Comment (abhishekagarwal87) [inline docs/configuration/index.md:622]: when the list is empty, no header is allowed, correct?

Comment (pranavbhole) [reply docs/configuration/index.md:622]: default value for druid.ingestion.http.allowedHeaders is empty list which means all headers are allowed.

Comment (pranavbhole) [reply docs/configuration/index.md:622]: if we we want to restrict headers then druid.ingestion.http.allowedHeaders should be non empty.

Comment (pranavbhole) [reply docs/configuration/index.md]: made changes to modify behavior. Default value for druid.ingestion.http.allowedHeaders is empty which allows no headers to pass.

File: processing/src/main/java/org/apache/druid/data/input/impl/HttpInputSource.java [hunk 2]
```diff
@@ -100,13 +100,15 @@ public static void throwIfInvalidProtocols(HttpInputSourceConfig config, List<UR
...
-      for (Map.Entry<String, String> entry : requestHeaders.entrySet()) {
-        if (!config.getAllowedHeaders().contains(StringUtils.toLowerCase(entry.getKey()))) {
-          throw InvalidInput.exception("Got forbidden header %s, allowed headers are only %s ",
-                                       entry.getKey(), config.getAllowedHeaders()
-          );
-        }
+    String message = null;
+    if (config.getAllowedHeaders().size() == 0 && requestHeaders.size() > 0) {
+      message = "You can set the property druid.ingestion.http.allowedHeaders in middle managers or peons to whitelist request headers";
+    }
```
Comment (abhishekagarwal87) [inline processing/src/main/java/org/apache/druid/data/input/impl/HttpInputSource.java:106]: this block can be removed while keeping the functionality just the same as before.

File: processing/src/main/java/org/apache/druid/data/input/impl/HttpInputSource.java [hunk 3]
```diff
@@ -100,13 +100,15 @@ public static void throwIfInvalidProtocols(HttpInputSourceConfig config, List<UR
...
+    for (Map.Entry<String, String> entry : requestHeaders.entrySet()) {
+      if (!config.getAllowedHeaders().contains(StringUtils.toLowerCase(entry.getKey()))) {
+        throw InvalidInput.exception("Got forbidden header %s, allowed headers are only %s. %s",
```
Comment (abhishekagarwal87) [inline processing/src/main/java/org/apache/druid/data/input/impl/HttpInputSource.java:109]:
```suggestion
        throw InvalidInput.exception("Got forbidden header [%s], allowed headers are only [%s]. You can control the allowed headers by updating druid.ingestion.http.allowedHeaders",
```

**Review by abhishekagarwal87 — APPROVED** [review]

**Pull request merged**